src/NuGet.Clients/VsExtension/VsCredentialProviderImporter.cs

```diff
@@ -59,39 +65,82 @@ public VsCredentialProviderImporter(DTE dte, Func<ICredentialProvider> fallbackP
                 throw new ArgumentNullException(nameof(fallbackProviderFactory));
             }
 
+            if (errorDelegate == null)
+            {
+                throw new ArgumentNullException(nameof(errorDelegate));
+            }
+
             _dte = dte;
+            _errorDelegate = errorDelegate;
             _fallbackProviderFactory = fallbackProviderFactory;
             _initializer = initializer ?? Initialize;
         }
 
-        [Import("VisualStudioAccountProvider", typeof(IVsCredentialProvider), AllowDefault = true)]
-        public IVsCredentialProvider ImportedProvider { get; set; }
+        [ImportMany(typeof(IVsCredentialProvider))]
```

Comment:
Unfortunately, this won't work in Dev15.  The export from Team Explorer is with a named contract:

[Export("VisualStudioAccountProvider", typeof(IVsCredentialProvider))]

Since the export includes both name and type, the import must as well.  ImportMany(typeof(IVsCredentialProvider)) will therefore load all of the IVsCredentialProviders except the one from TeamExplorer.

Safest will be to have both ImportMany(typeof(IVsCredentialProvider)) and ImportMany("VisualStudioAccountProvider", typeof(IVsCredentialProvider)) properties, and combine the two.  That way you get both the named and unamed exports, and can handle if someone else exports a provider with the same contract name.

If two extension both export the same named contract, and there is just a single named Import, neither will be imported.